civic-tech-club/src/components/BudgetVis.js

```diff
@@ -0,0 +1,226 @@
+import React, { useState, useEffect, useRef } from 'react';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import zoomPlugin from 'chartjs-plugin-zoom'; // Import the zoom plugin
+import data from '../data/budget.json';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+// Register the required Chart.js components and the zoom plugin
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler,
+  zoomPlugin // Register the zoom plugin here
+);
+
+// Helper to generate unique colors
+const COLORS = [
+  'rgba(75, 192, 192, 1)',
+  'rgba(255, 99, 132, 1)',
+  'rgba(54, 162, 235, 1)',
+  'rgba(255, 206, 86, 1)',
+  'rgba(153, 102, 255, 1)',
+  'rgba(255, 159, 64, 1)',
+];
+
+const BudgetVis = () => {
+  const [chartTitle, setChartTitle] = useState('');
+  const [chartData, setChartData] = useState({ labels: [], datasets: [] });
+  const chartRef = useRef(null); // Create a ref for the chart
+
+  useEffect(() => {
+    // This code will run once when the component loads
+    handleChartChange('Gross Operating Expenditures');
```

Comment:
if you put the initial values in the setState calls instead, you can reduce this useEffect down to just a `resetZoom` call. If that's not needed initially, we can get rid of the useEffect altogether.

Reply:
Thanks for this, I was able to refactor significantly and it's a lot cleaner. I'm open to any suggestions you have on how to make this better. I'm still new to react

Reply:
amazing. Good job!